daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/PhaseOneTest.scala

```diff
@@ -73,6 +128,95 @@ class PhaseOneTest extends AnyFreeSpec with Matchers with TableDrivenPropertyChe
+  private def scenBlock2 = (x: Expr) =>
+    EScenario(ScenarioBlock(ImmArray(Binding(None, ty, exp)), x))
+  private def scenCommit1 = (x: Expr) => EScenario(ScenarioCommit(x, exp, ty))
+  private def scenCommit2 = (x: Expr) => EScenario(ScenarioCommit(exp, x, ty))
+  private def scenMustFail1 = (x: Expr) => EScenario(ScenarioMustFailAt(x, exp, ty))
+  private def scenMustFail2 = (x: Expr) => EScenario(ScenarioMustFailAt(exp, x, ty))
+  private def scenPass = (x: Expr) => EScenario(ScenarioPass(x))
+  private def scenParty = (x: Expr) => EScenario(ScenarioGetParty(x))
+  private def scenEmbed = (x: Expr) => EScenario(ScenarioEmbedExpr(ty, x))
+  private def upure = (x: Expr) => EUpdate(UpdatePure(ty, x))
+  private def ublock1 = (x: Expr) =>
+    EUpdate(UpdateBlock(ImmArray(Binding(None, ty, x), Binding(None, ty, exp)), exp))
+  private def ublock2 = (x: Expr) =>
+    EUpdate(UpdateBlock(ImmArray(Binding(None, ty, exp), Binding(None, ty, x)), exp))
+  private def ublock3 = (x: Expr) =>
+    EUpdate(UpdateBlock(ImmArray(Binding(None, ty, exp), Binding(None, ty, exp)), x))
+  private def ucreate = (x: Expr) => EUpdate(UpdateCreate(tcon, x))
+  private def ucreateI = (x: Expr) => EUpdate(UpdateCreateInterface(tcon, x))
+  private def ufetch = (x: Expr) => EUpdate(UpdateFetch(tcon, x))
+  private def ufetchI = (x: Expr) => EUpdate(UpdateFetchInterface(tcon, x))
+  private def uexercise1 = (x: Expr) => EUpdate(UpdateExercise(tcon, choice, x, exp))
+  private def uexercise2 = (x: Expr) => EUpdate(UpdateExercise(tcon, choice, exp, x))
+  private def uexerciseI1 = (x: Expr) =>
+    EUpdate(UpdateExerciseInterface(tcon, choice, x, exp, exp, exp))
+  private def uexerciseI2 = (x: Expr) =>
+    EUpdate(UpdateExerciseInterface(tcon, choice, exp, x, exp, exp))
+  private def uexerciseI3 = (x: Expr) =>
+    EUpdate(UpdateExerciseInterface(tcon, choice, exp, exp, x, exp))
+  private def uexerciseI4 = (x: Expr) =>
+    EUpdate(UpdateExerciseInterface(tcon, choice, exp, exp, exp, x))
+  private def uexbykey1 = (x: Expr) => EUpdate(UpdateExerciseByKey(tcon, choice, x, exp))
+  private def uexbykey2 = (x: Expr) => EUpdate(UpdateExerciseByKey(tcon, choice, exp, x))
+  private def ufetchbykey = (x: Expr) => EUpdate(UpdateFetchByKey(RetrieveByKey(tcon, x)))
+  private def ulookupbykey = (x: Expr) => EUpdate(UpdateLookupByKey(RetrieveByKey(tcon, x)))
+  private def uembed = (x: Expr) => EUpdate(UpdateEmbedExpr(ty, x))
+  private def utrycatch1 = (x: Expr) => EUpdate(UpdateTryCatch(ty, x, varname, exp))
+  private def utrycatch2 = (x: Expr) => EUpdate(UpdateTryCatch(ty, exp, varname, x))
+
+  private def recCon1 = (x: Expr) => ERecCon(tapp, ImmArray((field, x), (field2, exp)))
+  private def recCon2 = (x: Expr) => ERecCon(tapp, ImmArray((field, exp), (field2, x)))
+  //private def recProj = (x: Expr) => ERecProj(tapp, field, x)
+  //private def recUpd1 = (x: Expr) => ERecUpd(tapp, field, x, exp)
+  //private def recUpd2 = (x: Expr) => ERecUpd(tapp, field, exp, x)
```

Comment:
What’s up with these?

Reply:
These testcases need additional setup to get working. They require a package containing a valid type definition.